src/pybind/mgr/dashboard_v2/frontend/src/app/shared/directives/password-button.directive.ts

```diff
@@ -0,0 +1,41 @@
+import { Directive, ElementRef, Input, HostListener, Renderer2 } from '@angular/core';
+
+@Directive({
+  selector: '[cdPasswordButton]'
+})
+export class PasswordButtonDirective {
+  private inputElement: any;
+  private iElement: any;
+
+  @Input('cdPasswordButton')
+  private inputId: string;
+
+  constructor(private el: ElementRef, private renderer: Renderer2) { }
+
+  ngOnInit() {
+    this.inputElement = document.getElementById(this.inputId);
+    this.iElement = this.renderer.createElement('i');
+    this.renderer.addClass(this.iElement, 'icon-prepend');
+    this.renderer.addClass(this.iElement, 'fa');
+    this.renderer.appendChild(this.el.nativeElement, this.iElement);
+    this.update();
+  }
+
+  private update() {
+    if (this.inputElement.type === "text") {
+      this.renderer.removeClass(this.iElement, 'fa-eye');
+      this.renderer.addClass(this.iElement, 'fa-eye-slash');
+    } else {
+      this.renderer.removeClass(this.iElement, 'fa-eye-slash');
+      this.renderer.addClass(this.iElement, 'fa-eye');
+    }
+  }
+
+  @HostListener('click')
+  onClick() {
+    // Modify the type of the input field.
+    this.inputElement.type = (this.inputElement.type === "password") ? "text" : "password";
+    // Update the button icon/tooltip.
+    this.update();
+  }
```

Comment:
Isn't it possible to use a template for this? With a template we don't have to add and remove classes and it's easier to read.

`<i #pwicon class="icon-prepend fa" [ngClass]="{'fa-eye':!isText(), 'fa-eye-slash':isText()}" (click)="changeType()">`

Reply:
Didn't find a way to do so.